channels/rdpgfx/server/rdpgfx_main.c

```diff
@@ -1361,16 +1361,25 @@ static UINT rdpgfx_server_receive_pdu(RdpgfxServerContext* context, wStream* s)
 	return error;
 }
 
+static BOOL rdpgfx_server_close(RdpgfxServerContext* context);
+
 static DWORD WINAPI rdpgfx_server_thread_func(LPVOID arg)
 {
 	RdpgfxServerContext* context = (RdpgfxServerContext*)arg;
+	WINPR_ASSERT(context);
+
 	RdpgfxServerPrivate* priv = context->priv;
 	DWORD status;
 	DWORD nCount = 0;
 	HANDLE events[8] = { 0 };
 	UINT error = CHANNEL_RC_OK;
 
-	events[nCount++] = priv->stopEvent;
+	WINPR_ASSERT(priv);
+	WINPR_ASSERT(priv->stopEvent);
+	WINPR_ASSERT(priv->channelEvent);
+
+	if (priv->ownThread)
```

Comment:
Maybe this could be an assertion here (since `rdpgfx_server_thread_func` would normally not run if `ownThread` is `FALSE` in the first place)?

Reply:
ok, updated. also refactored the thread setting to be compatible with other channels.